### ISSUE TITLE
Build: Skip contrib/pyln-grpc-proto ?

### ISSUE THREAD

**Issue by IngwiePhoenix** [ElementsProject/lightning#7282]
### Issue and Steps to Reproduce
Hello there,

I am trying to migrate my btc/cln stack into an incus Alpine 3.19 container, however I am facing issues while building v24.02.2: `grpc_tools.protoc` is missing, and while APK provides `py3-grpcio`, it does _not_ provide `py3-grpcio-tools`. Thus, this particular module can not be built.

Hence, I would like to skip it. Is there a way - like a configure flag - I can do that?

Thanks!

### `getinfo` output
None; my node is offline. Migrated from my previous setup with OpenWrt to Debian and Alpine in Incus as part of the Kubernetes-restructure of my network.

Kind regards,
Ingwie

**Comment by vincenzopalazzo** [comment]
Concept ack for this, compile cln on odroid is a very log process due the grpc depencencies

**Comment by daywalker90** [comment]
i believe there is a flag to disable building the rust crates, something like `./configure --disable-rust` iirc

**Comment by IngwiePhoenix** [comment]
Yes, you can en- or disable rust; but you can not do the same for the gRPC stuff, let alone for the Python modules - which I personally see no use for except perhaps for the built-in REST API, which in return doesn't need any of the gRPC stuff either as it is launched as a plugin that consumes STDIN/STDOUT.

**Comment by daywalker90** [comment]
the grpc stuff IS only from building the rust crate `cln-grpc`, i'm pretty sure i did not need a protoc compiler or grpc dependencies if i did not build that crate

**Comment by vincenzopalazzo** [comment]
@daywalker90 are you sure that it is not required also by pyln-client? I remember some hidden dependencies

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: Question issue with answers, inactive >1 year

If you believe this issue should remain open, please comment and we can reopen it.
